nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -893,10 +898,22 @@ def fwd_output_only_func(dataloader_iter, model):
                 if attention_mask is not None:
                     attention_mask = attention_mask.cuda()
                     attention_mask = attention_mask[0:1]
+            if self.mcore_gpt:
+                # if first step, then clear KV cache, otherwise reuse inference_paarms
+                if set_inference_key_value_memory[0].item():
+                    self.inference_params = InferenceParams(
```

Comment:
I don't personally like the name InferenceParam being passed around during model training. But since this is in core already, we can go ahead with it and probably change later if we fee like

Reply:
okay, since this is forward_only, should be good